lib/fungus_toast/games/grid.ex

```diff
@@ -49,32 +50,246 @@ defmodule FungusToast.Games.Grid do
     trunc(x_coordinate + grid_height_and_width * y_coordinate)
   end
 
-  def generate_growth_cycles(starting_grid, player_id_to_player_map, number_of_growth_cycles) do
-    #TODO get the live cells
-    #live_cells = :maps.filter fn _, v -> v.live end, v
+  def generate_growth_cycles(starting_grid, grid_size, player_id_to_player_map, number_of_growth_cycles) do
+    live_cells = :maps.filter(fn _, grid_cell -> grid_cell.live end, starting_grid)
 
-    #TODO return a map of index => %GridCell{} 
-    #Enum.each(live_cells, fn(x, y) -> {calculate_cell_growth(starting_grid, player_id_to_player_map, y) end)
+    #TODO return a map of index => %GridCell{} for newly grown and newly perished cells ONLY (i.e. changes only)
+    Enum.map(live_cells, fn{x, y} -> {x, calculate_cell_growth(starting_grid, grid_size, player_id_to_player_map, y)} end)
   end
 
-  def calculate_cell_growth(starting_grid, player_id_to_player_map, grid_cell) do
+  def calculate_cell_growth(starting_grid, grid_size, player_id_to_player_map, grid_cell) do
     #TODO get ALL surrounding cells (we'll need empty and live, dead, and empty ones)
-    #surrounding_cells = get_surrounding_cells(starting_grid, grid_cell.index)
+    #surrounding_cells = get_surrounding_cells(starting_grid, grid_size, grid_cell.index)
 
     #empty_surrounding_cells = :maps.filter(fn (_, v) -> v.empty end)
 
     #iterate over empty cells and calculate generate cells according ot the corresponding probabilities on player.*growth_chance. 
     #Return a list of newly generated GridCells
 
     #iterate over adjacent dead cells to calculate whether the cell is regenerated according to player.regeneration_chance
+    #add to the map of live cells if the cell was regenerated
+    #dead_surrounding_cells = :maps.filter(fn (_, v) -> !v.live end)
 
     #check if the cell dies from apoptosis, starvation, or mycotoxins
     #check_for_cell_death(grid_cell, surrounding_cells)
 
     #return a tuple which includes new split cells and regenerated cells, and and an indicator of whether the current cell died
   end
 
-  def get_surrounding_cells(grid, cell_index) do
-    #TODO
+  def get_surrounding_cells(grid, grid_size, cell_index) do
+    check_left = true
+    check_top = true
+    check_right = true
+    check_bottom = true
+
+    top_left_cell = nil
+    top_cell = nil
+    top_right_cell = nil
+    right_cell = nil
+    bottom_right_cell = nil
+    bottom_cell = nil
+    bottom_left_cell = nil
+    left_cell = nil
+
+    if(on_left_column(cell_index, grid_size)) do
```

Comment:
I would lean towards putting these in functions. From my understanding, `if`/`else` is not really idiomatic Elixir

Reply:
I agree. I really need to walk through this with someone to understand the alternative. In the meantime, I'm in git-r-done any way I can mode, and I'll refactor when someone has time to work through it with me.